### ISSUE TITLE
Parsing date from format `hh` gives valid true even if the hour is greater than 12

### ISSUE THREAD

**Issue by rohitkhatri** [moment/luxon#1655]
**Describe the bug**
Parsing time from string based on a `hh:mm a` format gives valid true even if the hour is greater than 12

**To Reproduce**
```typescript
import { DateTime } from "luxon";

const date = DateTime.fromFormat('13:00 AM', 'hh:mm a', {});
console.log(date);
console.log(date.isValid);
```

**Actual vs Expected behavior**
It should give invalid since the definition of `hh` is 12 hours clock

**Desktop (please complete the following information):**

- OS: MacOS M1
- Browser Chrome: 123
- Luxon version: 3.1.0
- Your timezone: Asia/Kolkata

**Comment by rohitkhatri** [comment]
Is there any workaround for this for now?

**Comment by dobon** [comment]
Kind of a late reply, but I think this is actually expected behavior. I checked how Moment handles this situation and it also considers "13:00 AM" to be valid (and representing "1:00 PM" or "13:00 PM" or just "13:00"). Basically the rule seems to be "if a 24-hour time is provided, trust that and ignore the meridiem whether it was provider or not.
```javascript
>moment('13:00 AM', 'hh:mm a').isValid();
>>true
>moment('13:00 AM', 'hh:mm a').format('h:mm a');
>>"1:00 PM"
>moment('13:00 PM', 'hh:mm a').format('h:mm a');
>>"1:00 PM"
```

practically speaking, if this is considered invalid for your application, then you'll need to add some additional logic to do that additional validation.

**Comment by diesieben07** [comment]
I agree that this is expected behavior. The general robustness principle of "be conservative in what you send, be liberal in what you accept" applies. 
However I would be open to a pull request which adds a `strict` option to `fromFormat` which enables stricter parsing.
